File: salt/lib/models/recipe/recipe.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:salt/models/food_category/food_category.dart';
import 'package:salt/models/ingredient/ingredient.dart';
import 'package:salt/models/user/user.dart';

part 'recipe.g.dart';

@JsonSerializable()
class Recipe {
  final String title;
  final String description;
  final String content;
  final String coverImgURL;
  final double readTime;
  final List<FoodCategory> categories;
  final List<Ingredient> ingredients;
  final User author;
  final DateTime createdAt;
  final DateTime updatedAt;

  Recipe({
    required this.title,
    required this.description,
    required this.content,
    required this.coverImgURL,
    required this.readTime,
    required this.categories,
    required this.ingredients,
    required this.author,
    required this.updatedAt,
    required this.createdAt,
  });

  factory Recipe.fromJson(Map<String, dynamic> json) {
    return _$RecipeFromJson(json);
  }

  Map<String, dynamic> toJson() => _$RecipeToJson(this);
}

File: deprecated/backend/src/models/recipe.ts
import { Document, model, Schema } from "mongoose";
import { MainUserDocument } from "./main-user";
import MongoPaging from "mongo-cursor-pagination";
import { IngredientSchema } from "./ingredient";

export type RecipeDocument = Document & {
  title: string;
  description: string;
  content: string;
  coverImgURL?: string;
  readTime: number;
  createdAt: Date;
  updatedAt: Date;
  ingredients: any;
  categories: any;
  author: MainUserDocument;
};

const RecipeSchema = new Schema<RecipeDocument>(
  {
    title: { type: String, required: true, trim: true, maxlength: 256 },
    description: { type: String, required: true, maxlength: 2048, trim: true },
    content: { type: String, required: true, trim: true },
    coverImgURL: { type: String },
    readTime: { type: Number, required: true },
    categories: {
      type: [{ type: Schema.Types.ObjectId, ref: "FoodCategory" }],
      required: true,
    },
    ingredients: {
      type: [IngredientSchema],
    },
    author: { type: Schema.Types.ObjectId, ref: "MainUser", required: true },
  },
  { timestamps: true }
);

RecipeSchema.plugin(MongoPaging.mongoosePlugin, { name: "paginateRecipe" });

const Recipe = model<RecipeDocument>("Recipe", RecipeSchema);
export default Recipe;
